nav2_core/include/nav2_core/goal_checker.hpp

```diff
@@ -65,7 +66,10 @@ class GoalChecker
    * @brief Initialize any parameters from the NodeHandle
    * @param nh NodeHandle for grabbing parameters
    */
-  virtual void initialize(const rclcpp_lifecycle::LifecycleNode::SharedPtr & nh) = 0;
+  virtual void initialize(
```

Comment:
Mhm, I like this. It gives them the flexibility to use it if they want, but they can also just ignore it in their definitions so its not a required element for algorithm developers. 

Nice!